### PR TITLE
Diary Page

### PR DESCRIPTION
**Pull request by Drapegnik**
Diary Page and fixes for Diary Block

Screen Shot 2020-01-15 at 11 33 31

### CODE REVIEW

**Comment by Drapegnik** [comment]
[Diary Fixes](https://trello.com/c/ZsQ7Z16J/408-diary-fixes)

**Comment by UladBohdan** [comment]
@Drapegnik please check out my questions in React code.

File: components/common/modal/modal.scss [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  top: 0;
+  width: 100%;
+
+  &__background {
+    background-color: $primary-black;
+    height: 100%;
+    opacity: 0.5;
+    position: fixed;
+    width: 100%;
+    z-index: 250; // more than .wir-sidebar
```
Comment (Drapegnik) [inline components/common/modal/modal.scss]: https://css-tricks.com/handling-z-index/#article-header-id-3

File: components/specials/diary/ArrowIcon.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+// TODO: to consider moving into babajka/assets repo.
+
+const ArrowIcon = ({ size, direction }) => (
+  <svg
+    width={size}
+    height={size}
+    viewBox="0 0 35 16"
+    fill="none"
+    style={{ transform: direction === 'right' ? 'rotate(180deg)' : '' }}
```
Comment (Drapegnik) [inline components/specials/diary/ArrowIcon.js:12]: 👍

Comment (UladBohdan) [reply components/specials/diary/ArrowIcon.js:12]: for future: svg also have `transform` operation, but it does not work in safari and iOS safari. but CSS-transform does work :)

**Comment by UladBohdan** [comment]
[experimental styling for cards from "Slavics languages" collection](https://trello.com/c/r8sTssvF/439-experimental-styling-for-cards-from-slavics-languages-collection)

**Comment by UladBohdan** [comment]
[Article Page: add collection card](https://trello.com/c/URDm2auN/432-article-page-add-collection-card)

**Comment by UladBohdan** [comment]
@Drapegnik Collection logic here is on you, reassigning

**Comment by Drapegnik** [comment]
depends on https://github.com/babajka/babajka-backend/pull/66

**Comment by UladBohdan** [comment]
[diary page social buttons block](https://trello.com/c/bIKyVq1k/453-diary-page-social-buttons-block)

**Comment by UladBohdan** [comment]
- [ ] TODO: `suggested articles` block on bottom

File: pages/diary.js [hunk 3]
```diff
@@ -22,31 +27,48 @@ const mapStateToProps = (state, { lang }) => ({
...
-      )}
-    </div>
-    <div className="diary-page__text">{fiberyRenderer(content)}</div>
-  </div>
-);
+}) => {
+  const router = useRouter();
+  return (
+    <>
+      <MetaTitle title={`${formatDate(date, DATE_FORMAT)}, ${name}`} type="article" />
```
Comment (Drapegnik) [inline pages/diary.js]: `${formatDate(date, DATE_FORMAT)}, ${name}` extract to var

File: styles/pages/diary.scss [hunk 4]
```diff
@@ -39,15 +39,21 @@
 
   &__arrows {
     line-height: 24px;
+    width: 80px;
 
-    &--left {
-      margin-right: 5px;
+    &--top {
+      //
```
Comment (Drapegnik) [inline styles/pages/diary.scss:44]: ?

File: pages/diary.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+              baseUrl={image}
+              mode="x"
+            />
+          )}
+        </div>
+        <div className="diary-page__text">{fiberyRenderer(content)}</div>
+        <div className="diary-page__share">
+          <ShareButtons
+            urlPath={router.asPath}
+            title={`${name} ${useLocalization('diary.wrote')} ${formatDate(date, DATE_FORMAT)}  `}
```
Comment (Drapegnik) [inline pages/diary.js]: to think about better wording

File: styles/pages/diary.scss (hunk 4 above)
Comment (UladBohdan) [reply styles/pages/diary.scss:44]: Well right now **top arrows** do not really need any specific css rules, but the class is in JS so I would like to keep it css file as well. For better understanding of the structure, I'd say.

File: pages/about.js [hunk 6]
```diff
@@ -4,73 +4,78 @@ import React from 'react';
...
-      </div>
-      <div className="about-page__description-text">
-        <Text id="about.section2-text" />
+      <div className="about-page__description-block">
+        <div className="about-page__description-header">
+          <Text id="about.section-team-header" />
+        </div>
+        <div className="about-page__description-text about-page__team">
+          {team.map(({ image, name, role }) => (
+            <div className="about-page__teammate teammate">
```
Comment (Drapegnik) [inline pages/about.js]: `<div key={name}`

File: pages/about.js [hunk 7]
```diff
@@ -4,73 +4,78 @@ import React from 'react';
...
+              <img className="teammate__pic" src={image} alt={name[lang]} />
+              <div className="teammate__name">{name[lang]}</div>
+              <div className="teammate__role">{role[lang]}</div>
```
Comment (Drapegnik) [inline pages/about.js]: lets use `localize` here, due to default lang fallback

File: pages/diary.js [hunk 8]
```diff
@@ -25,13 +25,35 @@ const mapStateToProps = (state, { lang }) => ({
...
+  if (dateIsToday(date)) {
+    shareText = SHARE_TEXT.today;
+    if (year) {
+      shareText = shareText.concat(SHARE_TEXT.year(year));
+    }
+  } else {
+    shareText = `${formatDate(date, DATE_FORMAT)} `;
+  }
+  return shareText.concat(`${name} ${localize('diary.wrote', 'be')}...`);
+};
```
Comment (Drapegnik) [inline pages/diary.js:41]:
```js
const getShareText = (date, name) => {
  const today = dateIsToday(date);
  const year = getYear(date);
  return [
      today ? 'Сёння' : formatDate(date, DATE_FORMAT),
      today && year && `, у ${year} годзе, `,
      `${name} ${localize('diary.wrote', 'be')}...`,
  ].filter(Boolean).join('');
}
```

File: pages/about.js (hunk 6 above)
Comment (UladBohdan) [reply pages/about.js]: is it ok `key` is cyrillic?

**Comment by UladBohdan** [comment]
[diary page articles block](https://trello.com/c/cUZgdXs6/454-diary-page-articles-block)

File: pages/about.js (hunk 6 above)
Comment (Drapegnik) [reply pages/about.js]: yep, the main thing is uniq

**Comment by UladBohdan** [comment]
[metadata for /about page](https://trello.com/c/xtzeTylZ/437-metadata-for-about-page)

**Review by UladBohdan — APPROVED** [review]

**Pull request merged**